tracing-opentelemetry/src/main/java/io/micronaut/tracing/DefaultOpenTelemetryFactory.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing;
+
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.context.annotation.Property;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import jakarta.inject.Singleton;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import static io.micronaut.runtime.ApplicationConfiguration.APPLICATION_NAME;
+
+/**
+ * Registers an OpenTelemetry bean.
+ *
+ * @author Nemanja Mikic
+ */
+@Factory
+public class DefaultOpenTelemetryFactory {
+
+    private static final String SERVICE_NAME_KEY = "otel.service.name";
+    private static final String DEFAULT_TRACES_EXPORTER = "otel.traces.exporter";
+    private static final String DEFAULT_METRICS_EXPORTER = "otel.metrics.exporter";
+    private static final String DEFAULT_LOGS_EXPORTER = "otel.logs.exporter";
+    private static final String NONE = "none";
+
+    /**
+     * The OpenTelemetry bean with default values.
+     * @param applicationName the application name from configuration
+     * @param otelProperties the configuration properties for the opentelemetry autoconfigure
+     * @return the OpenTelemetry bean with default values
+     */
+    @Singleton
+    @Primary
+    protected OpenTelemetry defaultOpenTelemetry(@Property(name = APPLICATION_NAME) String applicationName,
+                                                 @Property(name = "otel") Properties otelProperties) {
```

Comment:
These should be gated between a configuration properties type and should probably use the raw values. See https://github.com/micronaut-projects/micronaut-sql/blob/94f8a48ba7dc6f050b88e54b9a09af6f616a6fc9/hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/JpaConfiguration.java#L177-L186 for an example of how to do that.